### ISSUE TITLE
Can you create 2D Region Assignment similar to Region Prompt Control 

### ISSUE THREAD

**Issue by wardensc2** [hako-mikan/sd-webui-regional-prompter#226]
Hi Hako-Mikan

I love to use Region Prompt Control to generate image whether lantent couple, tile diffusion or your regional prompter.

However I find very difficult to use your 2D region assignment, it's a bit complicated

Can you create something more simple like Region Prompt Control where we can drag and drop, extend or reduce the prompt area like
Tile Diffusion Regional Promp:
https://github.com/pkuliyi2015/multidiffusion-upscaler-for-automatic1111

Something like this:

![Untitled-1](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/77379748/641a055c-c16c-4485-9c15-e622ae54dfa5)

Since Tile Diffusion author haven't updated the extension to support SDXL, we have to find the replacement and your extension is the very good extension but not simple to use. If we simplify the region assignment process it's much better for all users

Thank you for reading

**Comment by hako-mikan** [comment]
Indeed, I believe having such a UI would be useful, but I only started learning programming this year, so I haven't been able to study up to designing UI.

**Comment by arcusmaximus** [comment]
I've written such a UI for the Forge Couple extension -> https://github.com/arcusmaximus/sd-forge-couple/tree/draggable-box-ui
Feel free to take the code and integrate it into Regional Prompter. It shouldn't be too hard to do - copy the whole "typescript" folder (from which javascript\regions.js is generated) and make some adjustments to it and the Python code.

![321009112-fd056c62-4297-448e-9cf8-d7261f9b023c](https://github.com/hako-mikan/sd-webui-regional-prompter/assets/133129/9f195590-f9eb-4a10-bada-385ef72a82ec)

* [Create a Gradio textbox](https://github.com/arcusmaximus/sd-forge-couple/blob/draggable-box-ui/scripts/couple_ui.py#L41) for holding the region information. Set `visible = False` and `elem_id = 'some element ID'`.
* [Call the Javascript initialization function](https://github.com/arcusmaximus/sd-forge-couple/blob/draggable-box-ui/scripts/couple_ui.py#L61) when e.g. the "Enable" checkbox is changed.
* [Update the Javascript init function](https://github.com/arcusmaximus/sd-forge-couple/blob/draggable-box-ui/typescript/CustomRegionControl.ts#L18), giving it the correct elem_ids of the Gradio groupbox (where it should create its controls) and the textbox (where it should put the region information).
* The Javascript code will automatically detect changes in WebUI's image size parameters and update its canvas accordingly. The width/height textboxes and "visualize and make template" button are no longer necessary and can be removed.
* When an image gets generated, use [Region.deserialize](https://github.com/arcusmaximus/sd-forge-couple/blob/draggable-box-ui/scripts/couple_mapping.py#L20) to parse the regions from the hidden textbox.

**Comment by Yumeo0** [comment]
Is anyone implementing this? Because this is just objectively better to use than what we currently have.
